Installation issue: `spack install -v cpuinfo` fails on Windows

### Steps to reproduce the issue

```console
$ spack install -v cpuinfo
```
Everything builds correctly until the cpuinfo package itself. I followed https://spack.readthedocs.io/en/latest/getting_started.html#spack-on-windows

### Error message

<details><summary>Error message</summary>

<pre>
[12/40] C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1439~1.335\bin\Hostx64\x64\cl.exe  /nologo -DCPUINFO_LOG_LEVEL=0 -DCPUINFO_LOG_TO_STDIO=1 -D_CRT_SECURE_NO_WARNINGS -D_WIN32_WINNT=0x0601 -IC:\Users\ondrejcertik\repos\spack\.staging\cpuinfo-2023-11-04-hzmenkd\spack-src\include -IC:\Users\ondrejcertik\repos\spack\.staging\cpuinfo-2023-11-04-hzmenkd\spack-src\src /DWIN32 /D_WINDOWS /W3 /MD /O2 /Ob2 /DNDEBUG /showIncludes /FoCMakeFiles\cpuinfo_internals.dir\src\x86\vendor.c.obj /FdCMakeFiles\cpuinfo_internals.dir\cpuinfo_internals.pdb /FS -c C:\Users\ondrejcertik\repos\spack\.staging\cpuinfo-2023-11-04-hzmenkd\spack-src\src\x86\vendor.c
[13/40] C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1439~1.335\bin\Hostx64\x64\cl.exe  /nologo -DCPUINFO_LOG_LEVEL=0 -DCPUINFO_LOG_TO_STDIO=1 -D_CRT_SECURE_NO_WARNINGS -D_WIN32_WINNT=0x0601 -IC:\Users\ondrejcertik\repos\spack\.staging\cpuinfo-2023-11-04-hzmenkd\spack-src\include -IC:\Users\ondrejcertik\repos\spack\.staging\cpuinfo-2023-11-04-hzmenkd\spack-src\src /DWIN32 /D_WINDOWS /W3 /MD /O2 /Ob2 /DNDEBUG /showIncludes /FoCMakeFiles\cpuinfo_internals.dir\src\x86\init.c.obj /FdCMakeFiles\cpuinfo_internals.dir\cpuinfo_internals.pdb /FS -c C:\Users\ondrejcertik\repos\spack\.staging\cpuinfo-2023-11-04-hzmenkd\spack-src\src\x86\init.c
==> Error: ProcessError: Command exited with status 1:
    'ninja' '-j12' '-v'

6 errors found in build log:
     50    [31/40] C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1439~1.335\
           bin\Hostx64\x64\cl.exe  /nologo -DCPUINFO_SUPPORTED_PLATFORM=1 -IC:\
           Users\ondrejcertik\repos\spack\.staging\cpuinfo-2023-11-04-hzmenkd\s
           pack-src\include /DWIN32 /D_WINDOWS /W3 /MD /O2 /Ob2 /DNDEBUG /showI
           ncludes /FoCMakeFiles\cpu-info.dir\tools\cpu-info.c.obj /FdCMakeFile
           s\cpu-info.dir\ /FS -c C:\Users\ondrejcertik\repos\spack\.staging\cp
           uinfo-2023-11-04-hzmenkd\spack-src\tools\cpu-info.c
     51    [32/40] C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1439~1.335\
           bin\Hostx64\x64\cl.exe  /nologo -DCPUINFO_LOG_LEVEL=2 -DCPUINFO_LOG_
           TO_STDIO=1 -D_CRT_SECURE_NO_WARNINGS -D_WIN32_WINNT=0x0601 -Dcpuinfo
           _EXPORTS -IC:\Users\ondrejcertik\repos\spack\.staging\cpuinfo-2023-1
           1-04-hzmenkd\spack-src\src -IC:\Users\ondrejcertik\repos\spack\.stag
           ing\cpuinfo-2023-11-04-hzmenkd\spack-src\include /DWIN32 /D_WINDOWS
           /W3 /MD /O2 /Ob2 /DNDEBUG /showIncludes /FoCMakeFiles\cpuinfo.dir\sr
           c\x86\windows\init.c.obj /FdCMakeFiles\cpuinfo.dir\ /FS -c C:\Users\
           ondrejcertik\repos\spack\.staging\cpuinfo-2023-11-04-hzmenkd\spack-s
           rc\src\x86\windows\init.c
     52    [33/40] C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1439~1.335\
           bin\Hostx64\x64\cl.exe  /nologo  -IC:\Users\ondrejcertik\repos\spack
           \.staging\cpuinfo-2023-11-04-hzmenkd\spack-src\include -IC:\Users\on
           drejcertik\repos\spack\.staging\cpuinfo-2023-11-04-hzmenkd\spack-src
           \src /DWIN32 /D_WINDOWS /W3 /MD /O2 /Ob2 /DNDEBUG /showIncludes /FoC
           MakeFiles\cpuid-dump.dir\tools\cpuid-dump.c.obj /FdCMakeFiles\cpuid-
           dump.dir\ /FS -c C:\Users\ondrejcertik\repos\spack\.staging\cpuinfo-
           2023-11-04-hzmenkd\spack-src\tools\cpuid-dump.c
     53    [34/40] C:\Windows\system32\cmd.exe /C "cd . && C:\Users\ondrejcerti
           k\repos\spack\opt\spack\windows-windows10.0.22631-skylake\msvc-19.39
           .33523\cmake-3.29.2-dtiufemgmmiosopz5y2gdlapjx5isfhj\bin\cmake.exe -
           E vs_link_dll --intdir=CMakeFiles\cpuinfo.dir --rc=C:\PROGRA~2\WI3CF
           2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\10
           0226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MIB055~1\2022\COMMUN
           ~1\VC\Tools\MSVC\1439~1.335\bin\Hostx64\x64\link.exe /nologo CMakeFi
           les\cpuinfo.dir\src\api.c.obj CMakeFiles\cpuinfo.dir\src\cache.c.obj
            CMakeFiles\cpuinfo.dir\src\init.c.obj CMakeFiles\cpuinfo.dir\src\lo
           g.c.obj CMakeFiles\cpuinfo.dir\src\x86\init.c.obj CMakeFiles\cpuinfo
           .dir\src\x86\info.c.obj CMakeFiles\cpuinfo.dir\src\x86\vendor.c.obj
           CMakeFiles\cpuinfo.dir\src\x86\uarch.c.obj CMakeFiles\cpuinfo.dir\sr
           c\x86\name.c.obj CMakeFiles\cpuinfo.dir\src\x86\topology.c.obj CMake
           Files\cpuinfo.dir\src\x86\isa.c.obj CMakeFiles\cpuinfo.dir\src\x86\c
           ache\init.c.obj CMakeFiles\cpuinfo.dir\src\x86\cache\descriptor.c.ob
           j CMakeFiles\cpuinfo.dir\src\x86\cache\deterministic.c.obj CMakeFile
           s\cpuinfo.dir\src\x86\windows\init.c.obj  /out:cpuinfo.dll /implib:c
           puinfo.lib /pdb:cpuinfo.pdb /dll /version:0.0 /machine:x64 /INCREMEN
           TAL:NO  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib o
           le32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
     54    [35/40] C:\Windows\system32\cmd.exe /C "cd . && C:\Users\ondrejcerti
           k\repos\spack\opt\spack\windows-windows10.0.22631-skylake\msvc-19.39
           .33523\cmake-3.29.2-dtiufemgmmiosopz5y2gdlapjx5isfhj\bin\cmake.exe -
           E vs_link_exe --intdir=CMakeFiles\cpuid-dump.dir --rc=C:\PROGRA~2\WI
           3CF2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin
           \100226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MIB055~1\2022\COM
           MUN~1\VC\Tools\MSVC\1439~1.335\bin\Hostx64\x64\link.exe /nologo CMak
           eFiles\cpuid-dump.dir\tools\cpuid-dump.c.obj  /out:cpuid-dump.exe /i
           mplib:cpuid-dump.lib /pdb:cpuid-dump.pdb /version:0.0 /machine:x64 /
           INCREMENTAL:NO /subsystem:console  kernel32.lib user32.lib gdi32.lib
            winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.l
           ib advapi32.lib && cd ."
     55    [36/40] C:\Windows\system32\cmd.exe /C "cd . && C:\Users\ondrejcerti
           k\repos\spack\opt\spack\windows-windows10.0.22631-skylake\msvc-19.39
           .33523\cmake-3.29.2-dtiufemgmmiosopz5y2gdlapjx5isfhj\bin\cmake.exe -
           E vs_link_exe --intdir=CMakeFiles\isa-info.dir --rc=C:\PROGRA~2\WI3C
           F2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\1
           00226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MIB055~1\2022\COMMU
           N~1\VC\Tools\MSVC\1439~1.335\bin\Hostx64\x64\link.exe /nologo CMakeF
           iles\isa-info.dir\tools\isa-info.c.obj  /out:isa-info.exe /implib:is
           a-info.lib /pdb:isa-info.pdb /version:0.0 /machine:x64 /INCREMENTAL:
           NO /subsystem:console  cpuinfo.lib  kernel32.lib user32.lib gdi32.li
           b winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.
           lib advapi32.lib && cd ."
  >> 56    FAILED: isa-info.exe
     57    C:\Windows\system32\cmd.exe /C "cd . && C:\Users\ondrejcertik\repos\
           spack\opt\spack\windows-windows10.0.22631-skylake\msvc-19.39.33523\c
           make-3.29.2-dtiufemgmmiosopz5y2gdlapjx5isfhj\bin\cmake.exe -E vs_lin
           k_exe --intdir=CMakeFiles\isa-info.dir --rc=C:\PROGRA~2\WI3CF2~1\10\
           bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.
           0\x64\mt.exe --manifests  -- C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\T
           ools\MSVC\1439~1.335\bin\Hostx64\x64\link.exe /nologo CMakeFiles\isa
           -info.dir\tools\isa-info.c.obj  /out:isa-info.exe /implib:isa-info.l
           ib /pdb:isa-info.pdb /version:0.0 /machine:x64 /INCREMENTAL:NO /subs
           ystem:console  cpuinfo.lib  kernel32.lib user32.lib gdi32.lib winspo
           ol.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib adva
           pi32.lib && cd ."
     58    LINK: command "C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1439
           ~1.335\bin\Hostx64\x64\link.exe /nologo CMakeFiles\isa-info.dir\tool
           s\isa-info.c.obj /out:isa-info.exe /implib:isa-info.lib /pdb:isa-inf
           o.pdb /version:0.0 /machine:x64 /INCREMENTAL:NO /subsystem:console c
           puinfo.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.li
           b ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFES
           T:EMBED,ID=1" failed (exit code 1181) with the following output:
  >> 59    LINK : fatal error LNK1181: cannot open input file 'cpuinfo.lib'
     60
     61    [37/40] C:\Windows\system32\cmd.exe /C "cd . && C:\Users\ondrejcerti
           k\repos\spack\opt\spack\windows-windows10.0.22631-skylake\msvc-19.39
           .33523\cmake-3.29.2-dtiufemgmmiosopz5y2gdlapjx5isfhj\bin\cmake.exe -
           E vs_link_exe --intdir=CMakeFiles\cpu-info.dir --rc=C:\PROGRA~2\WI3C
           F2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\1
           00226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MIB055~1\2022\COMMU
           N~1\VC\Tools\MSVC\1439~1.335\bin\Hostx64\x64\link.exe /nologo CMakeF
           iles\cpu-info.dir\tools\cpu-info.c.obj  /out:cpu-info.exe /implib:cp
           u-info.lib /pdb:cpu-info.pdb /version:0.0 /machine:x64 /INCREMENTAL:
           NO /subsystem:console  cpuinfo.lib  kernel32.lib user32.lib gdi32.li
           b winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.
           lib advapi32.lib && cd ."
  >> 62    FAILED: cpu-info.exe
     63    C:\Windows\system32\cmd.exe /C "cd . && C:\Users\ondrejcertik\repos\
           spack\opt\spack\windows-windows10.0.22631-skylake\msvc-19.39.33523\c
           make-3.29.2-dtiufemgmmiosopz5y2gdlapjx5isfhj\bin\cmake.exe -E vs_lin
           k_exe --intdir=CMakeFiles\cpu-info.dir --rc=C:\PROGRA~2\WI3CF2~1\10\
           bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.
           0\x64\mt.exe --manifests  -- C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\T
           ools\MSVC\1439~1.335\bin\Hostx64\x64\link.exe /nologo CMakeFiles\cpu
           -info.dir\tools\cpu-info.c.obj  /out:cpu-info.exe /implib:cpu-info.l
           ib /pdb:cpu-info.pdb /version:0.0 /machine:x64 /INCREMENTAL:NO /subs
           ystem:console  cpuinfo.lib  kernel32.lib user32.lib gdi32.lib winspo
           ol.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib adva
           pi32.lib && cd ."
     64    LINK: command "C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1439
           ~1.335\bin\Hostx64\x64\link.exe /nologo CMakeFiles\cpu-info.dir\tool
           s\cpu-info.c.obj /out:cpu-info.exe /implib:cpu-info.lib /pdb:cpu-inf
           o.pdb /version:0.0 /machine:x64 /INCREMENTAL:NO /subsystem:console c
           puinfo.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.li
           b ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFES
           T:EMBED,ID=1" failed (exit code 1181) with the following output:
  >> 65    LINK : fatal error LNK1181: cannot open input file 'cpuinfo.lib'
     66
     67    [38/40] C:\Windows\system32\cmd.exe /C "cd . && C:\Users\ondrejcerti
           k\repos\spack\opt\spack\windows-windows10.0.22631-skylake\msvc-19.39
           .33523\cmake-3.29.2-dtiufemgmmiosopz5y2gdlapjx5isfhj\bin\cmake.exe -
           E vs_link_exe --intdir=CMakeFiles\cache-info.dir --rc=C:\PROGRA~2\WI
           3CF2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin
           \100226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MIB055~1\2022\COM
           MUN~1\VC\Tools\MSVC\1439~1.335\bin\Hostx64\x64\link.exe /nologo CMak
           eFiles\cache-info.dir\tools\cache-info.c.obj  /out:cache-info.exe /i
           mplib:cache-info.lib /pdb:cache-info.pdb /version:0.0 /machine:x64 /
           INCREMENTAL:NO /subsystem:console  cpuinfo.lib  kernel32.lib user32.
           lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.l
           ib comdlg32.lib advapi32.lib && cd ."
  >> 68    FAILED: cache-info.exe
     69    C:\Windows\system32\cmd.exe /C "cd . && C:\Users\ondrejcertik\repos\
           spack\opt\spack\windows-windows10.0.22631-skylake\msvc-19.39.33523\c
           make-3.29.2-dtiufemgmmiosopz5y2gdlapjx5isfhj\bin\cmake.exe -E vs_lin
           k_exe --intdir=CMakeFiles\cache-info.dir --rc=C:\PROGRA~2\WI3CF2~1\1
           0\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~
           1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC
           \Tools\MSVC\1439~1.335\bin\Hostx64\x64\link.exe /nologo CMakeFiles\c
           ache-info.dir\tools\cache-info.c.obj  /out:cache-info.exe /implib:ca
           che-info.lib /pdb:cache-info.pdb /version:0.0 /machine:x64 /INCREMEN
           TAL:NO /subsystem:console  cpuinfo.lib  kernel32.lib user32.lib gdi3
           2.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdl
           g32.lib advapi32.lib && cd ."
     70    LINK: command "C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1439
           ~1.335\bin\Hostx64\x64\link.exe /nologo CMakeFiles\cache-info.dir\to
           ols\cache-info.c.obj /out:cache-info.exe /implib:cache-info.lib /pdb
           :cache-info.pdb /version:0.0 /machine:x64 /INCREMENTAL:NO /subsystem
           :console cpuinfo.lib kernel32.lib user32.lib gdi32.lib winspool.lib
           shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.li
           b /MANIFEST:EMBED,ID=1" failed (exit code 1181) with the following o
           utput:
  >> 71    LINK : fatal error LNK1181: cannot open input file 'cpuinfo.lib'
     72
     73    [39/40] C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1439~1.335\
           bin\Hostx64\x64\cl.exe  /nologo -DCPUINFO_LOG_LEVEL=0 -DCPUINFO_LOG_
           TO_STDIO=1 -D_CRT_SECURE_NO_WARNINGS -D_WIN32_WINNT=0x0601 -IC:\User
           s\ondrejcertik\repos\spack\.staging\cpuinfo-2023-11-04-hzmenkd\spack
           -src\include -IC:\Users\ondrejcertik\repos\spack\.staging\cpuinfo-20
           23-11-04-hzmenkd\spack-src\src /DWIN32 /D_WINDOWS /W3 /MD /O2 /Ob2 /
           DNDEBUG /showIncludes /FoCMakeFiles\cpuinfo_internals.dir\src\x86\wi
           ndows\init.c.obj /FdCMakeFiles\cpuinfo_internals.dir\cpuinfo_interna
           ls.pdb /FS -c C:\Users\ondrejcertik\repos\spack\.staging\cpuinfo-202
           3-11-04-hzmenkd\spack-src\src\x86\windows\init.c
     74    ninja: build stopped: subcommand failed.

See build log for details:
  C:\Users\ondrejcertik\repos\spack\.staging\cpuinfo-2023-11-04-hzmenkd\spack-build-out.txt


</pre></details>


### Information on your system

x

### Additional information

x

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/cpuinfo/package.py
+++ b/var/spack/repos/builtin/packages/cpuinfo/package.py
@@ -3,6 +3,7 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+import sys
 from spack.package import *
 
 
@@ -32,12 +33,16 @@ class Cpuinfo(CMakePackage):
     depends_on("cmake@3.5:", type="build")
 
     def cmake_args(self):
+        # cpuinfo cannot produce a shared build with MSVC because it does not export
+        # any symbols
+        # cpuinfo CI builds "default" on Windows platform
+        build_type = "default" if sys.platform == "win32" else "shared"
         # https://salsa.debian.org/deeplearning-team/cpuinfo/-/blob/master/debian/rules
         return [
             self.define("CPUINFO_BUILD_UNIT_TESTS", False),
             self.define("CPUINFO_BUILD_MOCK_TESTS", False),
             self.define("CPUINFO_BUILD_BENCHMARKS", False),
-            self.define("CPUINFO_LIBRARY_TYPE", "shared"),
+            self.define("CPUINFO_LIBRARY_TYPE", build_type),
             self.define("CPUINFO_LOG_LEVEL", "error"),
             self.define("CMAKE_SKIP_RPATH", True),
         ]

--- a/var/spack/repos/builtin/packages/cpuinfo/package.py
+++ b/var/spack/repos/builtin/packages/cpuinfo/package.py
@@ -4,6 +4,7 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 import sys
+
 from spack.package import *
 
 

